wallet-new/src/Cardano/Wallet/Kernel/CoinSelection/Generic.hs

```diff
@@ -590,16 +590,30 @@ nLargestFromListBy f n = \xs ->
     dropOne (Just [_])    = Nothing
     dropOne (Just (_:as)) = Just as
 
--- | Proportionally divide the fee over each output
+-- | Proportionally divide the fee over each output.
+--
+-- There's a special 'edge-case' when the given input is a singleton list
+-- with one 0 coin. This is artifically created during input selection when
+-- the transaction's amount matches exactly the source's balance.
+-- In such case, we can't really compute any ratio for fees and simply return
+-- the whole fee back with the given change value.
```

Comment:
I would add a final section to this comment explaining why we do that: we return the whole fee because at the end of the day "it doesn't matter": as we are doomed anyway (we have depleted our Utxo to cover the amount, no chances we have money to cover the fee, too) this will be caught "one layer up" and the whole coin selection algorithm will fail accordingly, as it should.